nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -352,12 +352,15 @@ in {
       };
       dbhost = mkOption {
         type = types.nullOr types.str;
-        default = "localhost";
+        default = if cfg.database.createLocally then (if cfg.config.dbtype == "pgsql" then "/run/postgresql" else "localhost") else null;
+        defaultText = ''if config.services.nextcloud.database.createLocally
+          then (if config.services.nextcloud.config.dbtype == "pgsql" then "/run/postgresql"
+          else "localhost") else null'';
```

Comment:
This will look completely malformed in the rendered docs, see
![2023-04-13_13h02m09s_screenshot](https://user-images.githubusercontent.com/6025220/231739426-88a14340-27d3-4a4e-aaa8-b5306d5fc271.png)

Note: you can test the result by running `nix-build nixos/release.nix -A manualHTML.x86_64-linux` and then `xdg-open result/share/doc/nixos/index.html`.